aws_signing_helper/serve.go

```diff
@@ -0,0 +1,270 @@
+package aws_signing_helper
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+)
+
+const DefaultPort = 9911
+const LocalHostAddress = "127.0.0.1"
+
+var RefreshTime = time.Minute * time.Duration(5)
+
+type RefreshableCred struct {
+	AccessKeyId     string
+	SecretAccessKey string
+	Token           string
+	Expiration      time.Time
+}
+
+type Endpoint struct {
+	PortNum int
+	Server  *http.Server
+	TmpCred RefreshableCred
+}
+
+type SessionToken struct {
+	Expiration time.Time
+}
+
+const TOKEN_RESOURCE_PATH = "/latest/api/token"
+const SECURITY_CREDENTIALS_RESOURCE_PATH = "/latest/meta-data/iam/security-credentials/"
+
+const EC2_METADATA_TOKEN_HEADER = "x-aws-ec2-metadata-token"
+const EC2_METADATA_TOKEN_TTL_HEADER = "x-aws-ec2-metadata-token-ttl-seconds"
+const DEFAULT_TOKEN_TTL_SECONDS = "21600"
+
+const X_FORWARDED_FOR_HEADER = "X-Forwarded-For"
+
+const MAX_TOKENS = 256
+
+var mutex sync.Mutex
```

Comment:
Might be simpler to use the `Map` that's already in `sync`: https://pkg.go.dev/sync@go1.19.3#Map

Reply:
Thanks - I will switch to using `sync.Map` in my next revision.